packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -66,239 +66,209 @@ function getBlockContext( attributes, blockType ) {
-			}
-		}
-	}
-
-	/**
-	 * Called on mount or when a mismatch exists between the templates and
-	 * inner blocks, synchronizes inner blocks with the template, replacing
-	 * current blocks.
-	 */
-	synchronizeBlocksWithTemplate() {
-		const { template, block, replaceInnerBlocks } = this.props;
-		const { innerBlocks } = block;
-
-		// Synchronize with templates. If the next set differs, replace.
-		const nextBlocks = synchronizeBlocksWithTemplate(
-			innerBlocks,
-			template
-		);
-		if ( ! isEqual( nextBlocks, innerBlocks ) ) {
-			replaceInnerBlocks( nextBlocks );
-		}
-	}
-
-	updateNestedSettings() {
-		const {
-			blockListSettings,
-			allowedBlocks,
-			updateNestedSettings,
-			templateLock,
-			parentLock,
-			__experimentalCaptureToolbars,
-			__experimentalMoverDirection,
-		} = this.props;
+function useNestedSettingsUpdate(
+	clientId,
+	allowedBlocks,
+	templateLock,
+	captureToolbars,
+	__experimentalMoverDirection
+) {
+	const { updateBlockListSettings } = useDispatch( 'core/block-editor' );
+
+	const { blockListSettings, parentLock } = useSelect(
+		( select ) => {
+			const rootClientId = select(
+				'core/block-editor'
+			).getBlockRootClientId( clientId );
+			return {
+				blockListSettings: select(
+					'core/block-editor'
+				).getBlockListSettings( clientId ),
+				parentLock: select( 'core/block-editor' ).getTemplateLock(
+					rootClientId
+				),
+			};
+		},
+		[ clientId ]
+	);
 
+	useEffect( () => {
 		const newSettings = {
 			allowedBlocks,
 			templateLock:
 				templateLock === undefined ? parentLock : templateLock,
-			__experimentalCaptureToolbars:
-				__experimentalCaptureToolbars || false,
+			__experimentalCaptureToolbars: captureToolbars || false,
 			__experimentalMoverDirection,
 		};
 
 		if ( ! isShallowEqual( blockListSettings, newSettings ) ) {
-			updateNestedSettings( newSettings );
-		}
-	}
-
-	render() {
-		const {
-			enableClickThrough,
-			clientId,
-			hasOverlay,
-			__experimentalCaptureToolbars: captureToolbars,
-			forwardedRef,
-			block,
-			...props
-		} = this.props;
-		const { templateInProcess } = this.state;
-
-		if ( templateInProcess ) {
-			return null;
+			updateBlockListSettings( clientId, newSettings );
 		}
+	}, [
+		clientId,
+		blockListSettings,
+		allowedBlocks,
+		templateLock,
+		parentLock,
+		captureToolbars,
+		__experimentalMoverDirection,
+		updateBlockListSettings,
+	] );
+}
 
-		const classes = classnames( {
-			'has-overlay': enableClickThrough && hasOverlay,
-			'is-capturing-toolbar': captureToolbars,
-		} );
-
-		let blockList = (
-			<BlockList
-				{ ...props }
-				ref={ forwardedRef }
-				rootClientId={ clientId }
-				className={ classes }
-			/>
-		);
-
-		// Wrap context provider if (and only if) block has context to provide.
-		const blockType = getBlockType( block.name );
-		if ( blockType && blockType.providesContext ) {
-			const context = getBlockContext( block.attributes, blockType );
+function useInnerBlockTemplateSynchronization(
```

Comment:
Some JSDocs for these hooks would be greatly appreciated to understand their purpose and how they work.

Reply:
definitely!